scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -1000,17 +1012,69 @@ static bool load_balance_timer(void)
 {
 	struct llc_ctx *llcx;
 	int llc_id, j;
+	u64 ideal_sum, load_sum = 0, interactive_sum = 0;
 
-	if (nr_llcs == 1)
+	if (nr_llcs == 1 && !autoslice)
 		return false;
 
-	// Reset load values until next load balancing.
+	if (!autoslice)
+		goto reset_load;
+
+	bpf_for(llc_id, 0, nr_llcs) {
+		if (!(llcx = lookup_llc_ctx(llc_id)))
+			return false;
+
+		bpf_for(j, 0, nr_dsqs_per_llc) {
+			load_sum += llcx->dsq_load[j];
+			if (j == 0)
+				interactive_sum += llcx->dsq_load[j];
+		}
+	}
+	dbg("load %llu interactive %llu", load_sum, interactive_sum);
+
+	if (load_sum == 0 || load_sum < interactive_sum) {
+		scx_bpf_error("invalid load sum");
+		goto reset_load;
+	}
+
+	if (interactive_sum == 0) {
+		dsq_time_slices[0] = (11 * dsq_time_slices[0]) / 10;
```

Comment:
Kind of janky scaling, but when I tried to make it scale based on the proportion of load it was too jittery.